open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidationUtils.java

```diff
@@ -0,0 +1,123 @@
+    private static final ConsentCoreService consentCoreService = ConsentExtensionsDataHolder.getInstance()
+            .getConsentCoreService();
+
+    /**
+     * Method to check whether Idempotency handling is required.
+     *
+     * @return True if idempotency is required, else False.
+     */
+    static boolean isIdempotencyEnabledFromConfig() {
+        return parser.isIdempotencyValidationEnabled();
+    }
+
+    /**
+     * Method to retrieve the consent ids that have the idempotency key name and value as attribute.
+     *
+     * @param idempotencyKeyName     Idempotency Key Name
+     * @param idempotencyKeyValue    Idempotency Key Value
+     * @return   List of consent ids if available, else an empty list will be returned
+     */
+    static ArrayList<String> getConsentIdsFromIdempotencyKey(String idempotencyKeyName,
+                                                             String idempotencyKeyValue) {
+        try {
+            return consentCoreService.getConsentIdByConsentAttributeNameAndValue(
+                    idempotencyKeyName, idempotencyKeyValue);
+        } catch (ConsentManagementException e) {
+            log.debug("No consent ids found for the idempotency key value");
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Method to compare the client ID sent in the request and client id retrieved from the database.
+     *
+     * @param requestClientID     Client ID sent in the request
+     * @param dbClientID          client ID retrieved from the database
+     * @return   true if the client ID sent in the request and client id retrieved from the database are equal
+     */
+    static boolean isClientIDEqual(String requestClientID, String dbClientID) {
+        if (requestClientID == null) {
+            return false;
+        }
+        return requestClientID.equals(dbClientID);
+    }
+
+    /**
+     * Method to check whether difference between two dates is less than the configured time.
+     *
+     * @param createdTime Created Time of the request
+     * @return  true if the request is received within allowed time
+     */
+    static boolean isRequestReceivedWithinAllowedTime(long createdTime) {
+
+        if (createdTime == 0L) {
+            log.debug("Created time is of the previous request is not correctly set. Hence returning false");
+            return false;
+        }
+        String allowedTimeDuration = parser.getIdempotencyAllowedTime();
+        if (StringUtils.isNotBlank(allowedTimeDuration)) {
+            OffsetDateTime createdDate = OffsetDateTime.parse(toISO8601DateTime(createdTime));
+            OffsetDateTime currDate = OffsetDateTime.now(createdDate.getOffset());
+
+            long diffInMinutes = Duration.between(createdDate, currDate).toMinutes();
+            return diffInMinutes <= Long.parseLong(allowedTimeDuration);
+        } else {
+            log.error("Idempotency Allowed duration is configured in the system. Hence returning false");
```

Comment:
typo

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/473c6a6abe9cc7caaa346c896d00d9146ed1922c